### PR TITLE
Fixed Exploit and Possible Bypass

### PR DESCRIPTION
**Pull request by RH-Adv1c3**
- SellDrug minus was not checked if the amount is minus or not, which **could** cause issues in some builds
- Added Pickup Distance Check 90 distance from config position based upon the weed spawn field

### CODE REVIEW

**Comment by RH-Adv1c3** [comment]
Just noticed a small typo on my side, line 112 the src variable needs to be source instead.

File: server/main.lua [hunk 1]
```diff
@@ -13,6 +32,10 @@ AddEventHandler('esx_drugs:sellDrug', function(itemName, amount)
 		return
 	end
 
+	if amount < 0 then
```
Comment (Gellipapa) [inline server/main.lua:35]: If amount < 0  you should return. Ohh and 0 can sell?

File: server/main.lua [hunk 2]
```diff
@@ -80,43 +107,45 @@ end)
...
-      while outofbound == false and can do
-				if playersProcessingCannabis[source] == nil then
-					playersProcessingCannabis[source] = ESX.SetTimeout(Config.Delays.WeedProcessing , function()
-            if xCannabis.count >= 3 then
-              if xPlayer.canSwapItem('cannabis', 3, 'marijuana', 1) then
-                xPlayer.removeInventoryItem('cannabis', 3)
-                xPlayer.addInventoryItem('marijuana', 1)
-								xPlayer.showNotification(TranslateCap('weed_processed'))
-							else
+		if ValidateProcessCannabis(src) then
```
Comment (Gellipapa) [inline server/main.lua:112]: yes please fix this typo src to source

File: server/main.lua [hunk 3]
```diff
@@ -2,6 +2,25 @@ local playersProcessingCannabis = {}
...
+end
+
+local function ValidateProcessCannabis(src)
+	local ECoords = Config.CircleZones.WeedProcessing.coords
+	local PCoords = GetEntityCoords(GetPlayerPed(src))
+	local Dist = #(PCoords-ECoords)
+	if Dist <= 90 then return true end
+end
+
+local function BanPlayer(src,reason)
```
Comment (Gellipapa) [inline server/main.lua:19]: Rename method name, this is kick player, no ban player.

**Review by Gellipapa — CHANGES_REQUESTED** [review]
Please read my comments and fix problems.

**Comment by RH-Adv1c3** [comment]
Good catch, still sleeping half haha

**Review by Gellipapa — APPROVED** [review]
It's okey, did you test this?

**Comment by RH-Adv1c3** [comment]
> It's okey, did you test this?

No otherwise I would've found the typo and the variable name mismatch. 
I run a custom framework so I can't test it really easy

**Comment by Gellipapa** [comment]
It's okey , we need test this.

**Comment by RH-Adv1c3** [comment]
> It's okey , we need test this.

No problem, I found the script and thought I'd suggest some changes for more security.
You can see on my Profile that it's something I do more often

**Comment by Gellipapa** [comment]
@RH-Adv1c3 Hi! I was tested this pr, and it works but -1 you can't add input field, because Config you can add min and max value and if you write minus number in input, context menu return 1 all time and your print never show. One or two things still work interestingly but they are not your changes, we'll fix or rewrite the whole thing later.

**Comment by RH-Adv1c3** [comment]
> @RH-Adv1c3 Hi! I was tested this pr, and it works but -1 you can't add input field, because Config you can add min and max value and if you write minus number in input, context menu return 1 all time and your print never show. One or two things still work interestingly but they are not your changes, we'll fix or rewrite the whole thing later.

People can still trigger the server event directly with a mod menu, so anything that is not double checked in the server code is still possibly even though your client menu's are solid proof

**Comment by Gellipapa** [comment]
> > @RH-Adv1c3 Hi! I was tested this pr, and it works but -1 you can't add input field, because Config you can add min and max value and if you write minus number in input, context menu return 1 all time and your print never show. One or two things still work interestingly but they are not your changes, we'll fix or rewrite the whole thing later.
> 
> People can still trigger the server event directly with a mod menu, so anything that is not double checked in the server code is still possibly even though your client menu's are solid proof

Yes, that's why I accepted it because it's safe. Good job.

**Comment by RH-Adv1c3** [comment]
> > > @RH-Adv1c3 Hi! I was tested this pr, and it works but -1 you can't add input field, because Config you can add min and max value and if you write minus number in input, context menu return 1 all time and your print never show. One or two things still work interestingly but they are not your changes, we'll fix or rewrite the whole thing later.
> 
> > 
> 
> > People can still trigger the server event directly with a mod menu, so anything that is not double checked in the server code is still possibly even though your client menu's are solid proof
> 
> 
> 
> Yes, that's why I accepted it because it's safe. Good job.

I'll look further at it later today, because I know there is still one more minimal possible method to bypass the checks. So you'll get a new pull request today haha

**Pull request merged**